### PR TITLE
Host Tiles With GCOM

### LINKED ISSUE #59: R&D into self-hosting .pbf tiles for offline map usage
Currently the way that map-tiles are rendered is that they are either retrieved from an online API, or obtained from a separate local-hosted tile server, which ends up being a good amount of work involved . We can look into how these tiles could be instead directly hosted on GCOM instead.

Advantages:
- Less in-field setup work required, potentially less debugging in the case that the separate tile-hosting source doesn't work.
- One less process to run on the localhost machine, less complexity, greater chance of being able to bundle FE and BE together in a docker container or something like that.

Drawbacks: 
- Could be potentially very complex to setup. These map tiles are served in very special formats through layers of JSON metadata.
- Is the benefit-cost ratio worth it? Currently I'm thinking that we can spend a bit of time looking into this self-hosting solution for a couple weeks, but time is definitely better spent looking at something else if there is no good progress after e.g. a month

Alternatives:
- Maybe we could bundle GCOM BE, FE and the tile-hosting server all into one docker image for ease of setup.

### PR DESCRIPTION
**Pull request by Michael-R-Dickinson**
# Description

Added endpoints for fetching open street map data:
- Tile fetching
- osmbright style file
- metadata file
- fonts as glyphs

Resolves #59

## Type of change

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [ ]  Serve metadata file and style file
- [ ] Successfully and unsuccessfully serve tiles and glyphs

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging your code._

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/map_tiles/views.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        "tile_data",
+        str(z),
+        str(x),
+        f"{y}.pbf",
+    )
+
+    print("tiles_path", tiles_path)
+    if not os.path.exists(tiles_path):
+        response = FileResponse(
+            b"", content_type="application/x-protobuf", status=200  # or 204
```
Comment (jimgeng) [inline src/map_tiles/views.py:96]: 204 here to be consistent with what normally happens

File: src/pyproject.toml
[the remark this answers is not in the document]
Comment (Michael-R-Dickinson) [reply src/pyproject.toml:1]: We shouldn't have duplicates, it seems like a byproduct of somebody trying to poetry install in the wrong dir

File: src/map_tiles/apps.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    name = "map_tiles"
+
+    def ready(self):
+        """
+        This method is called when the Django app registry is fully populated.
+        It's a suitable place to perform initialization tasks.
+        """
+        # Prevent running initialization multiple times during development
+        if os.environ.get("RUN_MAIN") != "true":
+            return
```
Comment (jimgeng) [inline src/map_tiles/apps.py:16]: Probably can get rid of this, my current setup always returns this as None anyways.

**Review by jimgeng — COMMENTED** [review]
Currently glyphs aren't working, not sure what is causing this, since it does seem to be making the correct network requests (if you inspect you'll be able to see that it will make a request to `/Noto%20Sans%20Regular/0-255.pbf`

**Review by jimgeng — APPROVED** [review]
Functionality seems to be working (no glyphs still but that's fine).

I tested this by doing a fresh setup on my PC instead of my laptop, it works.

git in vscode is struggling to read this repo though, probably has to do with the large file sizes. Might have to look into that if it becomes a problem in the future

**Pull request merged**